### ISSUE TITLE
Improvements to CMake Windows instructions

### ISSUE THREAD

**Issue by rgov** [AlexeyAB/darknet#4826]
1. Run the "X64 Native Build Tools Command Prompt for VS 2019" shortcut in the Start menu.

2. Run these commands:

```
cd C:\path\to\darknet
mkdir build_release
cd build_release
cmake .. -GCMAKE_BUILD_TYPE=Release
cmake --build .
```

These commands in fact work on Linux also, because `cmake --build .` runs the build tool (`make` or Visual Studio) for you. No need to use the GUI.

The built products will appear in the `build_release/Debug/` subdirectory.

If the CUDA compiler is not detected automatically, try following [these instructions](https://devtalk.nvidia.com/default/topic/1049613/nsight-visual-studio-edition/visual-studio-2019-and-cuda-10-1/post/5327680/#5327680) to make sure Visual Studio knows how to build CUDA programs.

**Comment by cenit** [comment]
Thank you for your suggestions. In fact, maybe we can make those instruction clearer

Have you ever tried the ./build.ps1 script? It does all your steps and beyond :)

> Are these other build methods such as vcpkg needed? They are much more complicated.

What’s so difficult in `./vcpkg install darknet[full]`? I think that’s the easiest install path, which is also convenient to install CUDNN, Stb, pthreads and opencv in a consistent way (and automatically as dependencies from that command above)
As a cherry-pick bonus, you will have yolov2 and v3 weights ready, and also weights to start your custom training

**Comment by rgov** [comment]
I see now that the `build.ps1` script does use some of these commands. It does sound nice to have `vcpkg` install the dependencies for you.

However, I could not get the `build.ps1` script to work, although I am not typically a Windows user. For instance, under my Cygwin shell, the `pwsh` command was not available. 

If I am not mistaken the `CUDA_TOOLKIT_ROOT_DIR` and `CUDA_PATH` environment variables are not used by CMake anymore. I think you can set `CMAKE_CUDA_COMPILER` to the path to `nvcc`.

Feedback for the script:

1. CMake will automatically build in parallel, why offer `number_of_build_workers`?  The code is commented out anyway. If the user can uncomment this code, they could simply modify it to whatever `--parellel` option they want.

2. Rather than `use_ninja`, why not just allow them to provide an override for the `-G` option?

3. What is the advantage of `force_cpp_build`?

4. The changes between Release and Debug builds are minimal, why not make that an option as well, rather than a block of commented-out source?

**Comment by cenit** [comment]
0. For *nix environments (like cygwin) there is build.sh. Also, ninja is everything but undertested.... it’s the most used build system and it is also the way in which darknet is built inside Microsoft’s vcpkg continuous integration, which builds darknet from this repository and checks its consistency with all the dependents libraries on Windows, Linux, MacOS, UWP, in x86, x64, arm64 and arm variants....

1. Not everything is already ported to `enable_language(CUDA)`, some functions unfortunately are still available only under `find_package(CUDA)`. I wanted to write targeting the most modern CMake available, but unfortunately I had to call also the old find package for some functions. So we need both and we need also those extra env variables.

2. CMake does not build in parallel by default

3. Usually there is the native build chain (msbuild on Windows, make on *nix (apart from Xcode on Mac) OR Ninja everywhere. I don’t want to support anything else, so you can just enable ninja or not and then use the native way. No other -G options are supported.

4. To build also `.c` files using a c++ compiler. This is helpful to diagnose some problems

5. It looked like no one was using the debug build when building darknet from here, it was just a loss of time. Through vcpkg you have both configs anyway. Otherwise please un comment that part of the script
